src/components/App.jsx

```diff
@@ -0,0 +1,183 @@
+import { useCallback, useState } from "react";
+
+import { Api } from "../utils/Api";
+
+/*
+  Легенда:
+  Необходимо отобразить список ключей (произвольные строки), запрашиваемый с бэкенда.
+  При нажатии кнопки "Сгенерировать ключ" необходимо сгенерировать новый ключ (апи запрос) и отобразить его в списке.
+  Каждый ключ можно использовать. В таком случае он должен изменить свое визуальное отображение и увеличить счетчик использованных ключей.
+  Каждый ключ можно удалить. При удалении использованного ключа, счетчик использованных ключей уменьшается (счетчику важны только неудаленные использованные ключи)
+
+  Для поддержания актуального списка ключей необходимо обновлять список каждые 30 секунд. При получении от апи новых ключей считать их неиспользованными
+ */
+
+export function App() {
+    const [keys, setKeys] = useState(null);
+    const [isKeysRequested, setIsKeysRequested] = useState(false);
+
+    const [isLoading, setLoadingState] = useState(false);
+    const [countUsedKeys, setCountUsedKeys] = useState(0);
+
+    const toggleLoading = useCallback(() => {
+        setLoadingState(!isLoading);
+    }, [isLoading]);
+
+    const incrementUsedKeys = () => {
+        setCountUsedKeys((prevValue) => prevValue++);
+    }
+
+    const decrementUsedKeys = useCallback(() => {
+        setCountUsedKeys((prevValue) => prevValue--);
+    }, []);
+
+    const addKey = useCallback(async () => {
+        toggleLoading();
+
+        const key = await Api.generateKey();
+        setKeys((prevKeys) => prevKeys.push(key));
```

Comment:
Метод `.push` возвращает длину нового массива. Кажется здесь нужная другая инструкция добавления нового элемента в конец массива.
```suggestion
        setKeys((prevKeys) => [...prevKeys, key]);
```